### PR TITLE
some refactoring and better deadletter queue unavailable handling

### PR DESCRIPTION
**Pull request by neildsouth**
### Description

#Fixes
Problem if RabbitMq partly unavailable on startup

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [x] All tests passed locally.
- [ ] [Documentation comments](https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/language-specification/documentation-comments) included/updated.

### CODE REVIEW

File: src/Plugins/RabbitMQ/Tests/Integration/ReliabilityTest.cs [hunk 1]
```diff
@@ -139,8 +140,16 @@ public void GivenMessages_WhenPublished_SubscribeShallReceiveAndAckMessages()
                     Assert.True(_messages.TryUpdate($"{args.Message.MessageDescription}-{args.Message.MessageId}", 1, 0));
                     countDownEvent.Signal();
                 });
+
+                //subscriber.SubscribeAsync(topic, _topics[i], async Task (Common.MessageReceivedEventArgs args) =>
+                //{
+                //    subscriber.Acknowledge(args.Message);
+                //    Assert.True(_messages.TryUpdate($"{args.Message.MessageDescription}-{args.Message.MessageId}", 1, 0));
+                //    countDownEvent.Signal();
+                //});
```
Comment (woodheadio) [inline src/Plugins/RabbitMQ/Tests/Integration/ReliabilityTest.cs:149]: Please could you remove this commented-out code?

File: src/Plugins/RabbitMQ/Subscriber/RabbitMqMessageSubscriberService.cs [hunk 2]
```diff
@@ -438,5 +394,32 @@ private static MessageReceivedEventArgs CreateMessage(string topic, BasicDeliver
...
+                    _logger.DetectedInaccessibleNodeThatHousesDeadLetterQueue(queueName);
+                }
+                else
+                {
+                    return false;
+                }
+            }
+            catch (Exception ex)
+            {
+                var me = ex.Message;
```
Comment (woodheadio) [inline src/Plugins/RabbitMQ/Subscriber/RabbitMqMessageSubscriberService.cs:419]: If you just catch the exception without naming the exception, you should be able to drop this line.

Comment (woodheadio) [reply src/Plugins/RabbitMQ/Subscriber/RabbitMqMessageSubscriberService.cs:419]: Were there other types of exceptions that we wanted to catch here though?

Comment (neildsouth) [reply src/Plugins/RabbitMQ/Subscriber/RabbitMqMessageSubscriberService.cs]: just had a quick look and the exception you get when the queue doesn't exist is still a OperationInterruptedException, so I have removed the general catch now.

**Review by woodheadio — APPROVED** [review]

**Pull request merged**